### PR TITLE
Consider nested property paths when filtering Querydsl parameters.

### PR DESCRIPTION
**Pull request by lovrovrlec**
Fixes the regression reported in #2579.

Since GH-2572, `filterByJacksonVisibility` matches each request parameter against a flat map of
top-level Jackson field names. A nested key such as `address.street` is never a key in that map, so
it is dropped and the endpoint silently degrades to `findAll` — a filtered request returns every row.
Nested paths are otherwise fully supported: `QuerydslPredicateBuilder` resolves them via
`bindings.getPropertyPath(...)`, and they worked up to 5.0.5.

This resolves parameter names segment by segment, translating each segment through its own
`MappedJacksonProperties` so `@JsonProperty` renames keep working at depth, and rejecting the whole
parameter as soon as a segment is not exposed — extending the GH-2572 guarantee to nested paths
instead of weakening it.

Two tests, added in the first commit so the regression is visible on its own:
- `forwardsNestedPropertyPathsToQuerydsl` fails before the fix (the parameter never reaches Querydsl).
- `doesNotExposeJsonIgnoredPropertiesNestedInAssociations` passes before the fix only vacuously, and
  fails against a naive fix that forwards dotted keys unchecked — it pins the security property.

One deliberate decision worth your input: when a segment's owning type has no `MappedJacksonProperties`
(e.g. a map value), visibility cannot be established, so the parameter is rejected. This is stricter
than pre-5.0.5 behaviour, where such paths were forwarded and left to Querydsl's binding resolution to
reject. Happy to switch to forwarding them if you prefer the lenient variant.

### CODE REVIEW

File: spring-data-rest-tests/spring-data-rest-tests-mongodb/src/test/java/org/springframework/data/rest/webmvc/config/QuerydslAwareRootResourceInformationHandlerMethodArgumentResolverUnitTests.java [hunk 1]
```diff
@@ -180,6 +181,42 @@ void translatesJacksonRenamedPropertyToPersistentPropertyName() {
...
+		when(repositories.getRepositoryFor(User.class)).thenReturn(Optional.of(repository));
+
+		Map<String, String[]> parameters = Map.of("manager.ignored", new String[] { "candidate-value" });
+
+		ArgumentCaptor<MultiValueMap<String, String>> captor = ArgumentCaptor.captor();
+		when(builder.getPredicate(any(), captor.capture(), any())).thenReturn(mock(Predicate.class));
+
+		resolver.postProcess(parameter, invoker, User.class, parameters);
+
+		// User.manager is visible, but User.ignored is @JsonIgnore-annotated: the guarantee established for top-level
```
Comment (ruthst00) [inline spring-data-rest-tests/spring-data-rest-tests-mongodb/src/test/java/org/springframework/data/rest/webmvc/config/QuerydslAwareRootResourceInformationHandlerMethodArgumentResolverUnitTests.java]: Fix the test comment typo (`User.ignored` → `User.manager.ignored`)

File: spring-data-rest-tests/spring-data-rest-tests-mongodb/src/test/java/org/springframework/data/rest/webmvc/config/QuerydslAwareRootResourceInformationHandlerMethodArgumentResolverUnitTests.java [hunk 2]
```diff
@@ -180,6 +181,42 @@ void translatesJacksonRenamedPropertyToPersistentPropertyName() {
 		assertThat(forwarded.get("aliased")).containsExactly("value");
 	}
```
Comment (ruthst00) [inline spring-data-rest-tests/spring-data-rest-tests-mongodb/src/test/java/org/springframework/data/rest/webmvc/config/QuerydslAwareRootResourceInformationHandlerMethodArgumentResolverUnitTests.java:183]: - Consider adding a test for `@JsonProperty` rename at a nested level.
- Consider adding a test for the `properties == null` intermediate-type rejection path.

File: spring-data-rest-webmvc/src/main/java/org/springframework/data/rest/webmvc/config/QuerydslAwareRootResourceInformationHandlerMethodArgumentResolver.java [hunk 3]
```diff
@@ -123,26 +127,61 @@ private Optional<Pair<QuerydslPredicateExecutor<?>, Predicate>> getRepositoryAnd
...
+	 * filtering on nested properties (e.g. {@code address.street}). Returns {@literal null} if a segment does not refer
+	 * to a property Jackson exposes, so that a hidden property cannot be reached through an association either, or if
+	 * the type owning a segment is not backed by a {@link MappedJacksonProperties} instance, in which case visibility
+	 * cannot be established.
+	 *
+	 * @param domainType the type the leading segment is resolved against, must not be {@literal null}.
+	 * @param parameterName must not be {@literal null}.
+	 * @return the translated property path or {@literal null} if the parameter must not be forwarded to Querydsl.
+	 */
+	private @Nullable String resolvePath(Class<?> domainType, String parameterName) {
```
Comment (ruthst00) [inline spring-data-rest-webmvc/src/main/java/org/springframework/data/rest/webmvc/config/QuerydslAwareRootResourceInformationHandlerMethodArgumentResolver.java]: Guard against empty `parameterName` with an early return or a precondition check

File: spring-data-rest-webmvc/src/main/java/org/springframework/data/rest/webmvc/config/QuerydslAwareRootResourceInformationHandlerMethodArgumentResolver.java [hunk 4]
```diff
@@ -123,26 +127,61 @@ private Optional<Pair<QuerydslPredicateExecutor<?>, Predicate>> getRepositoryAnd
...
+
+		StringBuilder path = new StringBuilder(parameterName.length());
+		Class<?> owningType = domainType;
```
Comment (ruthst00) [inline spring-data-rest-webmvc/src/main/java/org/springframework/data/rest/webmvc/config/QuerydslAwareRootResourceInformationHandlerMethodArgumentResolver.java]: Add `Assert.notNull` guard for `domainType` — consistent with the rest of the class

File: spring-data-rest-webmvc/src/main/java/org/springframework/data/rest/webmvc/config/QuerydslAwareRootResourceInformationHandlerMethodArgumentResolver.java [hunk 5]
```diff
@@ -123,26 +127,61 @@ private Optional<Pair<QuerydslPredicateExecutor<?>, Predicate>> getRepositoryAnd
...
+
+		for (String segment : NESTED_PATH_SEPARATOR.split(parameterName)) {
```
Comment (ruthst00) [inline spring-data-rest-webmvc/src/main/java/org/springframework/data/rest/webmvc/config/QuerydslAwareRootResourceInformationHandlerMethodArgumentResolver.java]: Guard against empty `parameterName` with an early return or a precondition check

File: spring-data-rest-webmvc/src/main/java/org/springframework/data/rest/webmvc/config/QuerydslAwareRootResourceInformationHandlerMethodArgumentResolver.java [hunk 6]
```diff
@@ -123,26 +127,61 @@ private Optional<Pair<QuerydslPredicateExecutor<?>, Predicate>> getRepositoryAnd
...
+
+			MappedJacksonProperties properties = jacksonPropertiesLookup.apply(owningType);
+			PersistentProperty<?> property = properties == null ? null : properties.getPersistentProperty(segment);
```
Comment (ruthst00) [inline spring-data-rest-webmvc/src/main/java/org/springframework/data/rest/webmvc/config/QuerydslAwareRootResourceInformationHandlerMethodArgumentResolver.java]: Avoid calling `jacksonPropertiesLookup` twice for the root type.

**Review by ruthst00 — CHANGES_REQUESTED** [review]

File: spring-data-rest-tests/spring-data-rest-tests-mongodb/src/test/java/org/springframework/data/rest/webmvc/config/QuerydslAwareRootResourceInformationHandlerMethodArgumentResolverUnitTests.java (hunk 1 above)
Comment (lovrovrlec) [reply spring-data-rest-tests/spring-data-rest-tests-mongodb/src/test/java/org/springframework/data/rest/webmvc/config/QuerydslAwareRootResourceInformationHandlerMethodArgumentResolverUnitTests.java]: Fixed.

File: spring-data-rest-tests/spring-data-rest-tests-mongodb/src/test/java/org/springframework/data/rest/webmvc/config/QuerydslAwareRootResourceInformationHandlerMethodArgumentResolverUnitTests.java (hunk 2 above)
Comment (lovrovrlec) [reply spring-data-rest-tests/spring-data-rest-tests-mongodb/src/test/java/org/springframework/data/rest/webmvc/config/QuerydslAwareRootResourceInformationHandlerMethodArgumentResolverUnitTests.java:183]: Added translatesJacksonRenamedPropertyNestedInAssociation (needed a renamed property on Address, mirrors Profile.aliased) and rejectsNestedPathsBeneathTypesWithoutJacksonPropertyMapping.

File: spring-data-rest-webmvc/src/main/java/org/springframework/data/rest/webmvc/config/QuerydslAwareRootResourceInformationHandlerMethodArgumentResolver.java (hunk 5 above)
Comment (lovrovrlec) [reply spring-data-rest-webmvc/src/main/java/org/springframework/data/rest/webmvc/config/QuerydslAwareRootResourceInformationHandlerMethodArgumentResolver.java]: Blank segments now return null so the parameter is dropped like any unknown key. Went with early return over a precondition: names are client input, and an Assert would turn "?.street=x" into a 500. Splitting with limit -1 also rejects "address." instead of forwarding "address". Covered by dropsParameterNamesWithBlankSegments.

File: spring-data-rest-webmvc/src/main/java/org/springframework/data/rest/webmvc/config/QuerydslAwareRootResourceInformationHandlerMethodArgumentResolver.java (hunk 6 above)
Comment (lovrovrlec) [reply spring-data-rest-webmvc/src/main/java/org/springframework/data/rest/webmvc/config/QuerydslAwareRootResourceInformationHandlerMethodArgumentResolver.java]: Root MappedJacksonProperties is resolved once in filterByJacksonVisibility and handed to resolvePath. Nested segments still look up their owning type per parameter; a per-request cache seemed premature.

File: spring-data-rest-webmvc/src/main/java/org/springframework/data/rest/webmvc/config/QuerydslAwareRootResourceInformationHandlerMethodArgumentResolver.java (hunk 4 above)
Comment (lovrovrlec) [reply spring-data-rest-webmvc/src/main/java/org/springframework/data/rest/webmvc/config/QuerydslAwareRootResourceInformationHandlerMethodArgumentResolver.java]: resolvePath no longer takes domainType (see the double-lookup thread); both remaining parameters are now guarded with Assert.notNull.

**Comment by lovrovrlec** [comment]
@christophstrobl could you take a look at this when you next triage? It fixes a regression from GH-2572 (3dc76b8), reported in #2579.

**Review by ruthst00 — APPROVED** [review]
